packages/streaming-image-volume-loader/src/StreamingImageVolume.ts

```diff
@@ -83,18 +92,49 @@ export default class StreamingImageVolume extends ImageVolume {
     };
   }
 
-  private _hasLoaded = (): boolean => {
-    const { loadStatus, imageIds } = this;
-    const numFrames = imageIds.length;
+  protected initialize(): void {
+    this._createCornerstoneImageMetaData();
+  }
+
+  // The frameIndex is equal to imageIdIndex for 3D datasets
+  // but they are different for 4D datasets
+  protected imageIdIndexToFrameIndex(imageIdIndex: number): number {
+    return imageIdIndex % this.numFrames;
+  }
+
+  public get numFrames(): number {
+    return this.imageIds.length;
+  }
+
+  public getScalarDataArrays(): Array<Float32Array | Uint8Array> {
+    return [this.scalarData];
+  }
+
+  public getActiveScalarData(): Float32Array | Uint8Array {
```

Comment:
`getActiveScalarData` is weird name for 3D, can we just call it `getScalarData` and then for 4D it becomes `getScalarData(timepointIndex)` ?